Duplicati/Library/Backend/AliyunOSS/AliyunOSSBackend.cs

```diff
@@ -0,0 +1,364 @@
+
+namespace Duplicati.Library.Backend.AliyunOSS
+{
+    /// <summary>
+    /// Aliyun Object Storage Service(OSS) is a massive, secure, low-cost, and highly reliable cloud storage service, offering up to 99.995% service availability.It offers a variety of storage types to choose from, comprehensively optimizing storage costs.
+    /// en: https://www.alibabacloud.com/zh/product/object-storage-service
+    /// zh: https://www.aliyun.com/product/oss
+    /// </summary>
+    public class OSS : IBackend, IStreamingBackend, IRenameEnabledBackend
+    {
+        private static readonly string LOGTAG = Logging.Log.LogTagFromType<OSS>();
+
+        private const string OSS_REGION = "oss-region";
+        private const string OSS_BUCKET_NAME = "oss-bucket-name";
+        private const string OSS_ENDPOINT = "oss-endpoint";
+        private const string OSS_ACCESS_KEY_ID = "oss-access-key-id";
+        private const string OSS_ACCESS_KEY_SECRET = "oss-access-key-secret";
+
+        private AliyunOSSOptions _ossOptions;
+
+        public OSS()
+        { }
+
+        public OSS(string url, Dictionary<string, string> options)
+        {
+            _ossOptions = new AliyunOSSOptions();
+
+            var uri = new Utility.Uri(url?.Trim());
+            var prefix = uri.HostAndPath?.Trim()?.Trim('/')?.Trim('\\');
+
+            if (!string.IsNullOrEmpty(prefix))
+            {
+                _ossOptions.Path = prefix;
+            }
+
+            if (options.ContainsKey(OSS_REGION))
+            {
+                _ossOptions.Region = options[OSS_REGION];
+            }
+
+            if (options.ContainsKey(OSS_ACCESS_KEY_ID))
+            {
+                _ossOptions.AccessKeyId = options[OSS_ACCESS_KEY_ID];
+            }
+
+            if (options.ContainsKey(OSS_ACCESS_KEY_SECRET))
+            {
+                _ossOptions.AccessKeySecret = options[OSS_ACCESS_KEY_SECRET];
+            }
+
+            if (options.ContainsKey(OSS_BUCKET_NAME))
+            {
+                _ossOptions.BucketName = options[OSS_BUCKET_NAME];
+            }
+
+            if (options.ContainsKey(OSS_ENDPOINT))
+            {
+                _ossOptions.Endpoint = options[OSS_ENDPOINT];
+            }
+        }
+
+        private OssClient GetClient()
+        {
+            var endpoint = _ossOptions.Endpoint;
+            var accessKeyId = _ossOptions.AccessKeyId;
+            var accessKeySecret = _ossOptions.AccessKeySecret;
+            return new OssClient(endpoint, accessKeyId, accessKeySecret);
+        }
+
+        private OssClient GetClientV3()
+        {
+            var endpoint = _ossOptions.Endpoint;
+
+            var accessKeyId = _ossOptions.AccessKeyId;
+            var accessKeySecret = _ossOptions.AccessKeySecret;
+            return new OssClient(endpoint, accessKeyId, accessKeySecret, new ClientConfiguration()
+            {
+                UseNewServiceClient = false
+            });
+        }
+
+        //private Client GetClienV2()
```

Comment:
I am not a fan of leaving commented out code. Is there a reason we would ever want to support v2?

If there is a reason, the v2 and v3 code should be usable with an option (e.g., `--use-api-version=2`). In this case, I would suggest having a common interface, (e.g., `IOssClient`) and then having two separate files with the implementations.

If there is no good usecase for having v2 support, we should just delete the comments. The code can be dug out from Git should the need arise.

Reply:
Redundant and superfluous code indeed affects the cleanliness and readability of the code, so I have already cleaned it up and removed it